core/primitives/src/shard_layout.rs

```diff
@@ -75,10 +75,10 @@ type ShardSplitMap = Vec<Vec<ShardId>>;
 
 #[derive(serde::Serialize, serde::Deserialize, Clone, Debug, PartialEq, Eq)]
 pub struct ShardLayoutV1 {
-    /// num_shards = fixed_shards.len() + boundary_accounts.len() + 1
-    /// Each account and all sub-accounts map to the shard of position in this array.
-    fixed_shards: Vec<AccountId>,
```

Comment:
I'm not exactly sure how serde deserialize works but can we quickly verify whether deserializing an entry with `fixed_shards` entry doesn't crash/affect our struct?

Reply:
I'll add a small unit test for that. Looking at the docs - by default unknown fields should be ignored. 
https://serde.rs/container-attrs.html#deny_unknown_fields